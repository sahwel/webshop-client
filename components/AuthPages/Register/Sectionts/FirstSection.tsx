import React from 'react';
import { faEnvelope, faFaceGrin } from '@fortawesome/free-regular-svg-icons';
import Input from '../../../common/Form/Input/Input';

const FirstSection = () => {
  return (
    <>
      <p>Please enter some of your personal datas, then hit next!</p>
      <Input type="text" placeholder="Email" icon={faEnvelope} />
      <Input type="text" placeholder="First name" icon={faFaceGrin} />
      <Input type="text" placeholder="Last name" icon={faFaceGrin} />
    </>
  );
};

export default FirstSection;
