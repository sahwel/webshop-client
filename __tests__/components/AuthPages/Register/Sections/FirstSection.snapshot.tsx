import { render, renderHook, RenderHookResult } from '@testing-library/react';
import FirstSection from '@/components/AuthPages/Register/Sectionts/FirstSection';

describe('FirstSection Snapshot', () => {
  it('renders FirstSection unchanged', () => {
    const { container } = render(<FirstSection />);
    expect(container).toMatchSnapshot();
  });
});
