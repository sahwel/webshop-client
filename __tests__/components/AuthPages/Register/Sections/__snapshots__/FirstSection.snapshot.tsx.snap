// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstSection Snapshot renders FirstSection unchanged 1`] = `
<div>
  <p>
    Please enter some of your personal datas, then hit next!
  </p>
  <div
    class="input_container"
  >
    <input
      class="input"
      placeholder="Email"
      type="text"
    />
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-envelope input_icon icon-7"
      data-icon="envelope"
      data-prefix="far"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 128C0 92.65 28.65 64 64 64H448C483.3 64 512 92.65 512 128V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V128zM48 128V150.1L220.5 291.7C241.1 308.7 270.9 308.7 291.5 291.7L464 150.1V127.1C464 119.2 456.8 111.1 448 111.1H64C55.16 111.1 48 119.2 48 127.1L48 128zM48 212.2V384C48 392.8 55.16 400 64 400H448C456.8 400 464 392.8 464 384V212.2L322 328.8C283.6 360.3 228.4 360.3 189.1 328.8L48 212.2z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="input_container"
  >
    <input
      class="input"
      placeholder="First name"
      type="text"
    />
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-face-grin input_icon icon-7"
      data-icon="face-grin"
      data-prefix="far"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M349.5 308.4C368.2 303.1 385.4 320.4 374.1 336.5C350.4 374.6 306.3 399.1 255.9 399.1C205.6 399.1 161.5 374.6 136.9 336.5C126.5 320.4 143.7 303.1 162.3 308.4C191.3 315.1 222.8 318.8 255.9 318.8C289 318.8 320.6 315.1 349.5 308.4zM208.4 208C208.4 225.7 194 240 176.4 240C158.7 240 144.4 225.7 144.4 208C144.4 190.3 158.7 176 176.4 176C194 176 208.4 190.3 208.4 208zM304.4 208C304.4 190.3 318.7 176 336.4 176C354 176 368.4 190.3 368.4 208C368.4 225.7 354 240 336.4 240C318.7 240 304.4 225.7 304.4 208zM512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256zM256 48C141.1 48 48 141.1 48 256C48 370.9 141.1 464 256 464C370.9 464 464 370.9 464 256C464 141.1 370.9 48 256 48z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="input_container"
  >
    <input
      class="input"
      placeholder="Last name"
      type="text"
    />
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-face-grin input_icon icon-7"
      data-icon="face-grin"
      data-prefix="far"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M349.5 308.4C368.2 303.1 385.4 320.4 374.1 336.5C350.4 374.6 306.3 399.1 255.9 399.1C205.6 399.1 161.5 374.6 136.9 336.5C126.5 320.4 143.7 303.1 162.3 308.4C191.3 315.1 222.8 318.8 255.9 318.8C289 318.8 320.6 315.1 349.5 308.4zM208.4 208C208.4 225.7 194 240 176.4 240C158.7 240 144.4 225.7 144.4 208C144.4 190.3 158.7 176 176.4 176C194 176 208.4 190.3 208.4 208zM304.4 208C304.4 190.3 318.7 176 336.4 176C354 176 368.4 190.3 368.4 208C368.4 225.7 354 240 336.4 240C318.7 240 304.4 225.7 304.4 208zM512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256zM256 48C141.1 48 48 141.1 48 256C48 370.9 141.1 464 256 464C370.9 464 464 370.9 464 256C464 141.1 370.9 48 256 48z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
